pkg/gateway/services/stub.go

```diff
@@ -32,21 +33,35 @@ func (gws *GatewayService) GetOrCreateStub(ctx context.Context, in *pb.GetOrCrea
 		autoscaler.TasksPerContainer = uint(in.Autoscaler.TasksPerContainer)
 	}
 
+	taskPolicy := types.TaskPolicy{
+		MaxRetries: uint(in.Retries),
+		Timeout:    int(in.Timeout),
+	}
+
+	if in.TaskPolicy.Expiration > 0 {
```

Comment:
Can you clean up this casting? Feels like a lot of nested stuff

Reply:
reorganized